airflow/models.py

```diff
@@ -4062,7 +4072,11 @@ def verify_integrity(self, session=None):
             try:
                 dag.get_task(ti.task_id)
             except AirflowException:
-                if self.state is not State.RUNNING:
+                # TODO (bolke): a backfill can consider a subset of the original
+                # tasks of a whole dag but with the same name. If that is the case
+                # we should not mark tasks removed as we might not consider
+                # the whole dag.
+                if self.state is not State.RUNNING and not self.is_backfill:
```

Comment:
@mistercrunch we are not considering state changes of tasks here, but the state change of the whole dag run. The use case is clear to me, how to work it out on the DagRun level is bit different.